actions/postgresql_show_create_table_action.go

```diff
@@ -0,0 +1,466 @@
+	Attnotnull     bool
+	Attcollation   *string
+	Attidentity    string
+	Attstorage     string
+	Attstattarget  *string
+	ColDescription *string
+}
+
+type indexInfo struct {
+	Relname            string
+	IsPrimary          bool
+	IsUnique           bool
+	IsClustered        bool
+	IsValid            bool
+	Indrelid           string
+	PgGetIndexDef      *string
+	PgGetConstraintDef *string
+	Contype            *string
+	Condeferrable      *bool
+	Condeferred        *bool
+	Indisreplident     *bool
+	Reltablespace      int
+}
+
+type postgresqlShowCreateTableAction struct {
+	id     string
+	params *agentpb.StartActionRequest_PostgreSQLShowCreateTableParams
+}
+
+// NewPostgreSQLShowCreateTableAction creates PostgreSQL SHOW CREATE TABLE Action.
+// This is an Action that can run `\d+ table` command analog on PostgreSQL service with given DSN.
+func NewPostgreSQLShowCreateTableAction(id string, params *agentpb.StartActionRequest_PostgreSQLShowCreateTableParams) Action {
+	return &postgresqlShowCreateTableAction{
+		id:     id,
+		params: params,
+	}
+}
+
+// ID returns an Action ID.
+func (a *postgresqlShowCreateTableAction) ID() string {
+	return a.id
+}
+
+// Type returns an Action type.
+func (a *postgresqlShowCreateTableAction) Type() string {
+	return "postgresql-show-create-table"
+}
+
+// Run runs an Action and returns output and error.
+func (a *postgresqlShowCreateTableAction) Run(ctx context.Context) ([]byte, error) {
+	connector, err := pq.NewConnector(a.params.Dsn)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	db := sql.OpenDB(connector)
+	defer db.Close() //nolint:errcheck
+	buf := new(bytes.Buffer)
+
+	// Extract table id
+	tableID, err := a.printTableInit(ctx, buf, db)
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate table cells to be printed.
+	err = a.printColumnsInfo(ctx, buf, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Print indexes.
+	err = a.printIndexInfo(ctx, buf, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.printCheckConstraints(ctx, buf, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.printForeignKeyConstraints(ctx, buf, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.printReferencedBy(ctx, buf, db, tableID)
+	if err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+func (a *postgresqlShowCreateTableAction) printTableInit(ctx context.Context, w io.Writer, db *sql.DB) (string, error) {
+	var tableID, schema, relname string
+	row := db.QueryRowContext(ctx, `SELECT /* pmm-agent */  c.oid,
+	       n.nspname,
+	       c.relname
+	FROM pg_catalog.pg_class c
+	         LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+	WHERE c.relname = $1
+	  AND pg_catalog.pg_table_is_visible(c.oid)
+	ORDER BY nspname, relname;`, a.params.Table)
+	if err := row.Scan(&tableID, &schema, &relname); err != nil {
+		if err == sql.ErrNoRows {
+			return "", errors.WithStack(errors.Wrap(err, "Table not found"))
```

Comment:
`errors.Wrap` already calls `errors.WithStack`.

Use `errors.Wrap(err, "good message")` when you have a good message, or `errors.WithStack(err)` when you don't (or lazy).